### PR TITLE
[11.0][MIG] account_reconcile_rule_early_payment_discount

### PR DESCRIPTION
**Pull request by grindtildeath**
Migrated from : 
 - [ ] https://github.com/OCA/account-reconcile/pull/142

Depends on : 
 - [x] #209 
 - [x] https://github.com/OCA/account-payment/pull/184

### CODE REVIEW

**Comment by simahawk** [comment]
@grindtildeath pls do not include the other PR, update oca_dependencies.txt w/ a separated commit

**Comment by grindtildeath** [comment]
@simahawk Fixed

File: account_reconcile_rule_early_payment_discount/models/account_reconcile_rule.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                return False
+            elif not invoice.payment_term_id.early_payment_discount:
+                return False
+            else:
+                return self._check_early_payment_discount(
+                    statement_line, invoice, balance
+                )
+
+    @staticmethod
+    def _parse_date(str_date):
```
Comment (simahawk) [inline account_reconcile_rule_early_payment_discount/models/account_reconcile_rule.py]: this is not needed. Replace calls to `_parse_date` w/ `fields.Date.from_string`

File: account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+
+class TestEarlyPaymentDiscountRule(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Customer Partner
+        cls.partner = cls.env['res.partner'].create({
```
Comment (simahawk) [inline account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py:19]: tracking_disable

File: account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        cls.rule_model.search([]).unlink()
+
+        # Creation operation template and rule
+        cls.operation = cls.env['account.reconcile.model'].create({
+            'name': 'Unittest Early Payment Discount',
+            'label': 'Rounding',
+            'amount_type': 'percentage',
+            'amount': 100.0,
+        })
+
```
Comment (simahawk) [inline account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py]: too many empty lines

File: account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            'epd_discount': 2,
+        })
+
+        # Product and invoice with price = 1000
+        product_test = cls.env['product.product'].create({
+            'name': 'Unittest product',
+        })
+
+        cls.date_invoice = date(2016, 5, 10)
+        cls.invoice = cls.env['account.invoice'].create({
```
Comment (simahawk) [inline account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py:86]: tracking_disable

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by grindtildeath** [comment]
@tschanzt @p-tombez @simahawk @yvaucher 
Can you guys please pass by to get this merged ? :pray:

**Comment by i-vyshnevska** [comment]
test falling try to add post_install=True, 
seems they start falling from unmentioned dependency from purchase (purchase_warn)

**Comment by simahawk** [comment]
@grindtildeath see my comments on https://github.com/OCA/account-reconcile/pull/242

**Review by i-vyshnevska — APPROVED** [review]

**Review by simahawk — APPROVED** [review]

**Pull request closed without merging**